app/darwinInit.js

```diff
@@ -99,10 +114,18 @@ if (process.platform === 'darwin') {
       } catch (e) {
       }
 
-      // relaunch and append argument expected in:
+      // if user has the Muon version of Brave as their default browser, let's
+      // ask the user to make Brave Core their new default. Passing this extra
+      // argument will prompt the user to ensure they agree with the choice
+      let defaultBrowserArg = ''
+      if (getDefaultBrowser() === 'com.electron.brave') {
+        defaultBrowserArg = ' --make-default-browser'
+      }
+
+      // launch into freshly installed brave-core and append argument expected in:
       // https://github.com/brave/brave-browser/issues/1545
       console.log('Launching brave-core')
-      execSync(`open -a "${installedPath}/${appName}/" --args --upgrade-from-muon`)
+      execSync(`open -a "${installedPath}/${appName}/" --args --upgrade-from-muon${defaultBrowserArg}`)
```

Comment:
can we put space here instead of inside `defaultBrowserArg`?

Reply:
fixed